openlr/openlr_match_quality/openlr_assessment_tool/web_view.cpp

```diff
@@ -0,0 +1,138 @@
+#include "openlr/openlr_match_quality/openlr_assessment_tool/web_view.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+
+#include <QContextMenuEvent>
+#include <QDebug>
+#include <QMenu>
+#include <QMessageBox>
+#include <QTimer>
+
+namespace
+{
+bool IsLoginUrl(QString const & url)
+{
+  return url.contains("Login");
+}
+}  // namespace
+
+WebView::WebView(std::string const & url, std::string const & login, std::string const & paswd)
+  : QWebEngineView(nullptr), m_loadProgress(0), m_url(url), m_login(login), m_paswd(paswd)
+{
+  connect(this, &QWebEngineView::loadProgress, [this](int progress) { m_loadProgress = progress; });
+  connect(this, &QWebEngineView::loadFinished, [this](bool success) {
+    if (!success)
+    {
+      m_loadProgress = 0;
+    }
+  });
+
+  connect(this, &QWebEngineView::renderProcessTerminated,
+          [this](QWebEnginePage::RenderProcessTerminationStatus termStatus, int statusCode) {
+            QString status;
+            switch (termStatus)
+            {
+            case QWebEnginePage::NormalTerminationStatus:
+              status = tr("Render process normal exit");
```

Comment:
Are you sure that we need translations here?

Reply:
Done.